pkg/registration/hub/csr/reconciler.go

```diff
@@ -129,24 +137,25 @@ func (b *csrBootstrapReconciler) Reconcile(ctx context.Context, csr csrInfo, app
 		return reconcileContinue, nil
 	}
 
-	err := b.accpetCluster(ctx, clusterName)
+	err := b.acceptCluster(ctx, clusterName)
 	if errors.IsNotFound(err) {
-		// Current spoke cluster not found, could have been deleted, do nothing.
-		return reconcileStop, nil
+		// Current spoke cluster not found, it may not be created yet, requeue it until we cannot find its related CSR
+		b.eventRecorder.Warningf("ManagedClusterNotApproved", "spoke cluster %q is not found", clusterName)
+		return reconcileRequeue, nil
```

Comment:
So if we have a CSR but we cannot find the managed cluster we queue the request and will check again after some delay?

I expect that if this reconciler depends on a CSR and ManagedCluster, it will watch both resources. If the ManagedCluster is missing, you can do nothing, and once the ManagedCluster appears, you can approve the CSR.

Reply:
I will use a separated controller to handle cluster, and make this csr controller only handle the CSR ,  this will be more clear